src/stdlib_str2num.fypp

```diff
@@ -95,6 +95,7 @@ module stdlib_str2num
         integer(int8) :: p !! position within the number
         integer(int8)  :: stat !! error status
         !----------------------------------------------
+        if (.false.) v = mold ! removes unused-dummy-argument warning
```

Comment:
What does the compiler with this line? Does it remove it? What about the efficiency of adding this line of the whole procedure?

Reply:
I would refrain from such changes. In OO programming,  variables defined in abstract routines of the base type can easily be unused in some inherited types. Adding some lines just to suppress the warning is artificial and error-prone. I would rather advocate an option in fpm to build without the argument -Wunused-XXX

Reply:
Fully second leaving the unused dummy arguments as they are. In this simple case the compiler will remove the line, but it is indeed error prone and artificial. Unless one is 100% sure that the procedures signature can be safely changed, then the best solution in this case is to don't do anything.